### ISSUE TITLE
[QUESTION]: License change

### ISSUE THREAD

**Issue by rianquinn** [Bareflank/hypervisor#724]
I would like to start a discussion about moving from LGPL to MIT. LGPL prevents static linking and as a result, we have a lot of code in the base hypervisor specifically designed to support dynamic linking. In addition, we only support static linking for things like EFI, which means that anyone using the project that wanted to use EFI would have to open source which is not our intent. We don't want the license degrading the usefulness of the project. 

IMO, we should change to MIT. I do not think this will have any negative impacts on contributions, but I do think that it will greatly simplify the project. I would propose that we make this change as part of combining the EAPIs with the base hypervisor. 

Thoughts? Objections?

**Comment by connojd** [comment]
:+1:

**Comment by sskras** [comment]
Regarding MIT license.  I am not a developer in this project, but both hypevisors and open source licensing are interesting to me.  So:

https://writing.kemitchell.com/2017/03/29/OSS-Business-Perception-Report.html

As this open source attorney / law analyst writes, while MIT is very popular, quite short worded and very long living license:
https://writing.kemitchell.com/2016/09/21/MIT-License-Line-by-Line.html

... it still makes some difficulties for business to interpret it from a legal point of view.  Especially with regards to software patents.  Eg:
https://writing.kemitchell.com/2016/03/30/First-Read-of-the-Fair-Source-License.html

> Very broadly speaking, there are two “styles” of license grant language in open-source licenses.
> 
> One style is typical of the original “academic” licenses, notably the **MIT** and **BSD** license families. Those licenses use general words without specific legal meaning to give broad permission for software. Licenses of this style don’t usually mention the specific kinds of intellectual property—copyright, patent, trademark—being licensed.
> 
> I like to call the other style “contract-style” or “enterprisey”. **Apache 2.0** and **Eclipse 1.0** are examples. Those licenses use more technical language typical of professionally drafted—and often proprietary—software licenses. They address copyright, patent, and trademark rights specifically and separately.

Quoting the first article:
> 

> Apache-2.0 | ! | ? | Best permissive license. Professionally drafted. Explicit patent license.
> -- | -- | -- | --

So if the project will choose to increase its legal permissiveness, for the reasons above I suggest you considering **Apache License 2.0** .  

Just my .02€ !

**Comment by rianquinn** [comment]
@sskras Well, your comment sparked a lot of really good conversations internally. Basically, we landed on MIT. We have a CLA that addresses the patent issues that you discussed above. There are other issues with Apache like its redistribution clause and incompatibility with GPL v2. MIT keeps things really simple while the CLA ensures that patents are covered during the time the contribution is made.

**Comment by rianquinn** [comment]
FYI, when we do the license change (which should happen very shortly here as we attempt to merge in the Boxy changes), I will also update the CLA to use the following:
https://www.apache.org/licenses/icla.pdf

This way, our license is still MIT, but we leverage an already existing CLA that addresses the issues identified above in plain English.

**Comment by rianquinn** [comment]
Complete
